### ISSUE TITLE
Expose new parameters from DecoderHints in Python interface?

### ISSUE THREAD

**Issue by duanqn** [zxing-cpp/zxing-cpp#422]
Looking at https://github.com/zxing-cpp/zxing-cpp/blob/master/wrappers/python/zxing.cpp, it seems that some new parameters are not exposed in Python (e.g. downscale threshold). Should we keep the interface updated?

**Comment by axxel** [comment]
I'd like to do that only if there is a specific request/need.

**Comment by duanqn** [comment]
Yeah one of my colleagues wants to tune the downscale threshold with the Python interface. Is that a valid request? 🤔

**Comment by duanqn** [comment]
I can also try to expose it and submit a PR, if you think it's OK :)

**Comment by axxel** [comment]
I see ;). I suggest you add the parameter in your code, let you colleagues experiment with it and if it makes a real difference to them, we'll continue the discussion?

**Comment by duanqn** [comment]
It is useful because we are cropping the images based on some NN barcode detector. We need to lower the `downscale_threshold` otherwise it does not work well. PR #427

**Comment by axxel** [comment]
I see. Thanks for the explanation. So this means you have input that is lower in both width and height than those 500 pixels, this input does not get recognized without downscaling. Can you maybe provide a sample image together with the downscale threshold you found to be usable for your application? A potentially better alternative to adding the parameter might be to simply lower the default threshold, therefore I'm asking...

**Comment by duanqn** [comment]
This is what I learned during our exploration -- whether an image needs to be downscaled does not depend on its size, but depends on the size of barcodes inside it. For example, suppose we have a 2000x2000 image, ZXing-cpp will downscale it to recognize the barcodes. If we use this NN detector to crop a 200x200 area which contains a QR code, it may still need to be downscaled.
In our use case, the barcodes occupy >90% of the cropped image area, so the downscale threshold should be set differently. We don't have enough experience right now to guide the parameter setting for the general cases.

**Comment by duanqn** [comment]
I can prepare a sample image next week.

**Comment by duanqn** [comment]
@axxel have you got any chance to try the sample?

**Comment by axxel** [comment]
I did not have a look (until now). Unfortunately, I don't see how your sample shows the problem you are having. The size of this image is clearly large enough to trigger the downscaling and that then finds both `0200001545` symbols. I was looking for a sample of less than 500 pixels width/height where an additional downscaling is required to detect the symbol.

Are you saying that the mentioned fix for #425 does already fix your problem?

**Comment by duanqn** [comment]
> Are you saying that the mentioned fix for https://github.com/zxing-cpp/zxing-cpp/issues/425 does already fix your problem?

I haven't tested. But I imagine it would not solve the entire problem. For example if the original image is 2000x2000, and there is a QR code 400x400 large, and it requires downscaling 1 time to be recognized. Now if we crop this QR code area out, it would not trigger any downscaling (threshold being 500) so the code will not be recognized. To match the exact behaviour, I'll need to first calculate how many times the original image would have been downscaled and set the downscale threshold on the cropped image, so they are downscaled for the same levels.

However, as our exploration work has concluded, I'm OK with not exposing those parameters through the Python API too. I just thought it would be nice to expose them because some people don't want to modify and recompile this project when they are experimenting with neural network models. They just want something like `import zxing` and hack with the parameters.

**Comment by axxel** [comment]
> and there is a QR code 400x400 large, and it requires downscaling 1 time to be recognized

That is exactly the kind of input I'm interested in seeing. Can you provide that?
